### ISSUE TITLE
Need ability to convert 3D coordinates

### ISSUE THREAD

**Issue by whitefractal** [georust/proj#150]
`Proj::convert` assumes 2D coordinates, and passes `z: 0.0` into `proj_sys` (which handles 3D just fine). 

This is insufficient for conversions to/from (e.g.) ECEF coordinates, where the z coordinate is critical even for locations on the ground. 

Then need to access the private `c_proj` makes it impossible to define a trait with a minimally-modified `convert` method which takes a 3D point, passes all three values in/out of proj_sys, and returns a 3D point.

Some provision for 3D coords within this crate would be extremely helpful please :-)

**Comment by lnicola** [comment]
Yeah, we should definitely support 3-4D transforms, it's just a question of API design.

**Comment by lnicola** [comment]
CC https://github.com/georust/proj/issues/121

**Comment by urschrei** [comment]
I agree but I don't have time, so (and I mean this in the most constructive way) if you want it, you'll almost certainly have to write it.

**Comment by balazsdukai** [comment]
I have a fork that I adapted for 3D coordinate conversion, specifically, because I needed to reproject to ECEF @whitefractal .

Here is the [proj.rs](https://github.com/3DGI/tyler/blob/master/src/proj.rs) fork. [Here](https://github.com/3DGI/tyler/blob/85109336b206b566183cb36a9e57401a60132f99/src/formats.rs#L1267) is an example how I use it.

In my adaptation, I did the minimum necessary for my use case. However, because the current [Coord](https://github.com/georust/proj/blob/main/src/proj.rs#L64-L83) is 2D only, I'm not sure if adding additional members (eg `z`), would break anything downstream. What do you think @lnicola @urschrei ?

But if you don't see big hurdles in the API, I'll implement the 3D across proj and send a PR.

**Comment by Glatzel** [comment]
I created my own proj rust wrapper that support 2d,3d,4d conversion. Most proj features are work in progress but I think the basic conversion functions can work as expected.
https://github.com/Glatzel/pyxis/blob/d1c8f2fb25108668133fa4f181c329f498cae8f7/rust/crates/proj/src/extension/conversion.rs
